src/ast2ram/seminaive/ClauseTranslator.cpp

```diff
@@ -636,9 +709,25 @@ std::vector<ast::Atom*> ClauseTranslator::getAtomOrdering(const ast::Clause& cla
 
     // get the imposed order, and change it to start at zero
     const auto& order = orders.at(version);
-    std::vector<unsigned int> newOrder(order->getOrder().size());
+    auto sz = order->getOrder().size();
+    std::vector<unsigned int> newOrder(sz);
+
+    // assign dominatedHead/dominatingHead index in case
+    // a query plan has been specified for a subsumptive
+    // clause.
+    if (isA<ast::SubsumptiveClause>(clause)) {
+        for (std::size_t i = 0; i < sz; ++i) {
+            if (order->getOrder()[i] == 0) {
+                dominatedHead = newOrder[0];
```

Comment:
I may misunderstand this... `newOrder` has not been initialised yet (gets initialised on line 728). So it looks to me this just has the effect of setting `dominatedHead` and `dominatingHead` to 0 (assuming a zero-initialised vector)?

Reply:
Indeed - we need a test case for subsumptive clauses with a query plan.